rustls/src/client/tls13.rs

```diff
@@ -146,6 +149,10 @@ pub(super) fn handle_server_hello(
 
     let shared_secret = our_key_share.complete(&their_key_share.payload.0)?;
 
+    // TODO(@cpu): Handle confirmation of ECH.
```

Comment:
As discussed a while ago, I question the value of leaving these loose ends in main branch history (relative to squashing more things here).

Reply:
I think this is the most awkward join point and it could be resolved by squashing the "ech: construct ECH client hello" commit with the subsequent "ech: confirm acceptance of ECH offer" commit.

I'll do that in the rebase that applies your feedback. :+1:

Reply:
> could be resolved by squashing the "ech: construct ECH client hello" commit with the subsequent "ech: confirm acceptance of ECH offer" commit.

I think we'll actually end up wanting three commits joined: the config setup, the hello construct, and the acceptance confirm.

Reply:
I ended up squashing the majority of the implementation into one commit that:

* adds the config elements
* generates the hello
* confirms acceptance
* confirms acceptance w/ HRR

The GREASE commit remains separate since it didn't require any intermediate TODOs.